coderd/database/queries/notificationsinbox.sql

```diff
@@ -0,0 +1,71 @@
+-- name: GetUnreadInboxNotificationsByUserID :many
+--
+SELECT * FROM inbox_notifications WHERE
+	user_id = @user_id AND
+	read_at IS NULL AND
+	(@id_opt::UUID = '00000000-0000-0000-0000-000000000000'::UUID OR id > @id_opt::UUID)
+	ORDER BY created_at DESC
+	LIMIT (COALESCE(NULLIF(@limit_opt :: INT, 0), 25));
+
+-- name: GetInboxNotificationsByUserID :many
+SELECT * FROM inbox_notifications WHERE
+	user_id = @user_id AND
+	(@id_opt::UUID = '00000000-0000-0000-0000-000000000000'::UUID OR id > @id_opt::UUID)
+	ORDER BY created_at DESC
+	LIMIT (COALESCE(NULLIF(@limit_opt :: INT, 0), 25));
+
+-- name: GetInboxNotificationsByUserIDFilteredByTemplatesAndTargets :many
+-- Fetches inbox notifications for a user filtered by templates and targets
+-- param user_id: The user ID
+-- param templates: The template IDs to filter by - the template_id = ANY(@templates::UUID[]) condition checks if the template_id is in the @templates array
+-- param targets: The target IDs to filter by - the targets @> COALESCE(@targets, ARRAY[]::UUID[]) condition checks if the targets array (from the DB) contains all the elements in the @targets array
+SELECT * FROM inbox_notifications WHERE
+	user_id = @user_id AND
+	template_id = ANY(@templates::UUID[]) AND
+	targets @> COALESCE(@targets, ARRAY[]::UUID[]) AND
+	(@id_opt::UUID = '00000000-0000-0000-0000-000000000000'::UUID OR id > @id_opt::UUID)
+	ORDER BY created_at DESC
+	LIMIT (COALESCE(NULLIF(@limit_opt :: INT, 0), 25));
+
+-- name: GetUnreadInboxNotificationsByUserIDFilteredByTemplatesAndTargets :many
+-- param user_id: The user ID
+-- param templates: The template IDs to filter by - the template_id = ANY(@templates::UUID[]) condition checks if the template_id is in the @templates array
+-- param targets: The target IDs to filter by - the targets @> COALESCE(@targets, ARRAY[]::UUID[]) condition checks if the targets array (from the DB) contains all the elements in the @targets array
```

Comment:
As a caller, it's not really clear to me what kind of UUID a target should be without looking at example usages. Could this be reworded slightly to be clearer?